### PR TITLE
Add TCK results Jakarta Connectors 2.1

### PR DESCRIPTION
**Pull request by smillidge**
Signed-off-by: Steve Millidge <steve.millidge@payara.fish>

### CODE REVIEW

File: docs/website/src/main/resources/certifications/jakarta-connectors/2.1/TCK-Results.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+# 2.1 Certification Request
+
+- [x] Organization Name ("Organization") and, if applicable, URL: <br/>
+  Eclipse Foundation
+- [x] Product Name, Version and download URL (if applicable): <br/>
+  [GlassFish 7.0.0-M4](https://repo1.maven.org/maven2/org/glassfish/main/distributions/glassfish/7.0.0-M4/glassfish-7.0.0-M4.zip)
+- [x] Specification Name, Version and download URL: <br/>
+  [Jakarta Connectors 2.1](https://jakarta.ee/specifications/connectors/2.1/)
+- [x] TCK Version, digital SHA-256 fingerprint and download URL: <br/>
+  [Jakarta Connectors TCK 2.1.0](https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee10/staged/eftl/jakarta-connectors-tck-2.1.0.zip), SHA-256: 6b044e8361e1fd6e5ab46f3dfc3850332b23a192a1b872f878b3380aa9561bcc
```
Comment (ivargrimstad) [inline docs/website/src/main/resources/certifications/jakarta-connectors/2.1/TCK-Results.md:17]: SHA value should be: f8c018fe56f19899a780df3f8d10c93555996861038b7da03eee5421574d4b9b

Comment (hs536) [reply docs/website/src/main/resources/certifications/jakarta-connectors/2.1/TCK-Results.md:17]: it was changed again ( [jakarta-connectors-tck-2.1.0-tckinfo.txt](https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee10/staged/eftl/jakarta-connectors-tck-2.1.0-tckinfo.txt) ). Would it be a good idea to append a download date until the final version of TCK is released?

Comment (ivargrimstad) [reply docs/website/src/main/resources/certifications/jakarta-connectors/2.1/TCK-Results.md:17]: The important thing with regards to ratifying the spec and claiming compatibility for an implementation is that it is the one and only TCK that is used for the CCR. It has to be the same SHA that is published on the website. A CCR with a different SHA would not be valid.

**Comment by gurunrao** [comment]
Please note: connector TCK has failures with GF 7 nightly build and JDK 17.
Detailed run log can be found at https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck/job/master/1750/artifact/connectortck.log
CI run test result details: https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck/job/master/1750/
Test run archive with GF logs:  https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck/job/master/1750/artifact/connectors-tck-results.tar.gz

File: docs/website/src/main/resources/certifications/jakarta-connectors/2.1/TCK-Results.md (hunk 1 above)
Comment (gurunrao) [reply docs/website/src/main/resources/certifications/jakarta-connectors/2.1/TCK-Results.md:17]: its recommended to use promoted tck (https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee10/promoted/eftl/) for certification and ballot purpose, promotion can be done by sending mail to tck-dev(post stable tck results are got).
I have disabled the connector TCK build at CI, so the new connector TCK is not staged.

**Comment by smillidge** [comment]
I will look to get the TCK promoted so the sha stabilises. I ran the staged TCK against the milestone as the CI not nightly. I don't have time to investigate nightly failures.

**Comment by gurunrao** [comment]
failures with GF M4 JDK 17 - https://ci.eclipse.org/jakartaee-tck/job/guru/job/jakartaee-tck-build-run/job/master/32/#showFailuresLink

**Comment by smillidge** [comment]
Does it also fail on JDK 11? My concern is that many tests failing is likely every test failing which sounds pretty fundamental  in GlassFish although no change has been made to Connectors. Whereas I am trying to get Connectors to ballot.

**Comment by gurunrao** [comment]
> Does it also fail on JDK 11? My concern is that many tests failing is likely every test failing which sounds pretty fundamental in GlassFish although no change has been made to Connectors. Whereas I am trying to get Connectors to ballot.

All connector tests are passing with GF 7 and JDK 11. Logs for JDK11 + GF 7  runs - https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck/job/master/1749/artifact/connectortck.log

**Comment by arjantijms** [comment]
@gurunrao Can you see whether it's the TCK itself that's not JDK 17 compatible there, or GF?

**Comment by dmatej** [comment]
> failures with GF M4 JDK 17 - https://ci.eclipse.org/jakartaee-tck/job/guru/job/jakartaee-tck-build-run/job/master/32/#showFailuresLink

```
/home/jenkins/agent/workspace/u_jakartaee-tck-build-run_master/connectors-tck/bin/xml/impl/glassfish/deploy.xml:200: 
Deployment timeout reached - 220 seconds.
```

```
+ ./asadmin start-domain
Picked up JAVA_TOOL_OPTIONS: -Xmx6G
Waiting for domain1 to start ....
Successfully started the domain : domain1
domain  Location: /home/jenkins/agent/workspace/u_jakartaee-tck-build-run_master/glassfish7/glassfish/domains/domain1
Log File: /home/jenkins/agent/workspace/u_jakartaee-tck-build-run_master/glassfish7/glassfish/domains/domain1/logs/server.log
Admin Port: 4848
Command start-domain executed successfully.
+ ./asadmin create-jvm-options -Djava.security.manager
Picked up JAVA_TOOL_OPTIONS: -Xmx6G
Cannot read admin cache file for cache/localhost_4848/create-jvm-options
Created 1 option(s)
Command create-jvm-options executed successfully.
+ ./asadmin restart-domain
Picked up JAVA_TOOL_OPTIONS: -Xmx6G
Cannot read admin cache file for cache/localhost_4848/__locations
Cannot read admin cache file for cache/localhost_4848/_get-runtime-info
Cannot read admin cache file for cache/localhost_4848/restart-domain
Timed out waiting for the server to restart
Command restart-domain failed.
+ ./asadmin stop-domain
Picked up JAVA_TOOL_OPTIONS: -Xmx6G
Cannot read admin cache file for cache/localhost_4848/__locations
CLI306: Warning - The server located at /home/jenkins/agent/workspace/u_jakartaee-tck-build-run_master/glassfish7/glassfish/domains/domain1 is not running.
Command stop-domain executed successfully.
```

What we know is that restart-domain sometimes collides on ports and file system, because we still did not fix it. I will get to it soon, but for know it would be better to use just stop-domain and start-domain.

**Comment by gurunrao** [comment]
> @gurunrao Can you see whether it's the TCK itself that's not JDK 17 compatible there, or GF?

sure arjan.

**Comment by gurunrao** [comment]
TCK wars fails to deploy times out in JDK 17, error msg is as follows:

`  [gf.echo] Added sun xml file(s) to /home/jenkins/agent/workspace/u_jakartaee-tck-build-run_master/connectors-tck/dist/com/sun/ts/tests/connector/xa/lifecycle/xa_lifecycle_connectorservlet_vehicle_web.war
     [copy] Copying 1 file to /home/jenkins/agent/workspace/u_jakartaee-tck-build-run_master/connectors-tck/tmp
  [gf.echo] Looking at /home/jenkins/agent/workspace/u_jakartaee-tck-build-run_master/connectors-tck/dist/com/sun/ts/tests/connector/xa/security/xa_security_connectorservlet_vehicle_web.war
      [jar] Updating jar: /home/jenkins/agent/workspace/u_jakartaee-tck-build-run_master/connectors-tck/dist/com/sun/ts/tests/connector/xa/security/xa_security_connectorservlet_vehicle_web.war
  [gf.echo] Added sun xml file(s) to /home/jenkins/agent/workspace/u_jakartaee-tck-build-run_master/connectors-tck/dist/com/sun/ts/tests/connector/xa/security/xa_security_connectorservlet_vehicle_web.war
     [copy] Copying 1 file to /home/jenkins/agent/workspace/u_jakartaee-tck-build-run_master/connectors-tck/tmp
  [gf.echo] Looking at /home/jenkins/agent/workspace/u_jakartaee-tck-build-run_master/connectors-tck/dist/com/sun/ts/tests/connector/xa/workmgt/xa_workmgt_connectorservlet_vehicle_web.war
      [jar] Updating jar: /home/jenkins/agent/workspace/u_jakartaee-tck-build-run_master/connectors-tck/dist/com/sun/ts/tests/connector/xa/workmgt/xa_workmgt_connectorservlet_vehicle_web.war
  [gf.echo] Added sun xml file(s) to /home/jenkins/agent/workspace/u_jakartaee-tck-build-run_master/connectors-tck/dist/com/sun/ts/tests/connector/xa/workmgt/xa_workmgt_connectorservlet_vehicle_web.war
  [gf.echo] 22 archive(s) to deploy
     [copy] Copying 22 files to /home/jenkins/agent/workspace/u_jakartaee-tck-build-run_master/glassfish7/glassfish/domains/domain1/autodeploy
  [gf.echo] Deploying archive:  /home/jenkins/agent/workspace/u_jakartaee-tck-build-run_master/connectors-tck/dist/com/sun/ts/tests/connector/annotations/anno/annotations_connectorservlet_vehicle_web.war

BUILD FAILED
/home/jenkins/agent/workspace/u_jakartaee-tck-build-run_master/connectors-tck/bin/xml/ts.top.import.xml:293: The following error occurred while executing this line:
/home/jenkins/agent/workspace/u_jakartaee-tck-build-run_master/connectors-tck/bin/xml/impl/glassfish/deploy.xml:78: The following error occurred while executing this line:
/home/jenkins/agent/workspace/u_jakartaee-tck-build-run_master/connectors-tck/bin/xml/impl/glassfish/deploy.xml:152: The following error occurred while executing this line:
/home/jenkins/agent/workspace/u_jakartaee-tck-build-run_master/connectors-tck/bin/xml/impl/glassfish/deploy.xml:200: Deployment timeout reached - 220 seconds.` 

Also when we check for GF process, the GF is not up an running, As suggested by dmatej, issue related to restart will be one of the cause. The GF logs too don't have errors related to deployment.

**Comment by smillidge** [comment]
@ivargrimstad SHA updated

**Review by ivargrimstad — APPROVED** [review]

**Comment by ivargrimstad** [comment]
Great! As soon as we get this merged, Jakarta Connectors 2.1 is ready for the ballot to start

**Review by arjantijms — APPROVED** [review]

**Pull request merged**